x-pack/legacy/plugins/ml/server/routes/modules.ts

```diff
@@ -0,0 +1,240 @@
+) {
+  const dr = new DataRecognizer(context);
+  return dr.setupModuleItems(
+    moduleId,
+    prefix,
+    groups,
+    indexPatternName,
+    query,
+    useDedicatedIndex,
+    startDatafeed,
+    start,
+    end,
+    jobOverrides,
+    datafeedOverrides,
+    savedObjectsClient
+  );
+}
+
+function dataRecognizerJobsExist(context: RequestHandlerContext, moduleId: string) {
+  const dr = new DataRecognizer(context);
+  return dr.dataRecognizerJobsExist(moduleId);
+}
+
+/**
+ * Recognizer routes.
+ */
+export function dataRecognizer({
+  xpackMainPlugin,
+  router,
+  elasticsearchPlugin,
+  savedObjects,
+}: RouteInitialization) {
+  /**
+   * @apiGroup DataRecognizer
+   *
+   * @api {get} /api/ml/modules/recognize/:indexPatternTitle Recognize index pattern
+   * @apiName RecognizeIndex
+   * @apiDescription Returns the list of modules that matching the index pattern.
+   *
+   * @apiParam {String} indexPatternTitle Index pattern title.
+   */
+  router.get(
+    {
+      path: '/api/ml/modules/recognize/{indexPatternTitle}',
+      validate: {
+        params: schema.object({
+          indexPatternTitle: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { indexPatternTitle } = request.params;
+        const results = await recognize(context, indexPatternTitle);
+
+        return response.ok({ body: results });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup DataRecognizer
+   *
+   * @api {get} /api/ml/modules/get_module/:moduleId Get module
+   * @apiName GetModule
+   * @apiDescription Returns module by id.
+   *
+   * @apiParam {String} [moduleId] Module id
+   */
+  router.get(
+    {
+      path: '/api/ml/modules/get_module/{moduleId?}',
+      validate: {
+        params: schema.object({
+          moduleId: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        let { moduleId } = request.params;
+        if (moduleId === '') {
+          // if the endpoint is called with a trailing /
+          // the moduleId will be an empty string.
+          moduleId = undefined;
+        }
+        const results = await getModule(context, moduleId);
+
+        return response.ok({ body: results });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup DataRecognizer
+   *
+   * @api {post} /api/ml/modules/setup/:moduleId Setup module
+   * @apiName SetupModule
+   * @apiDescription Created module items.
+   *
+   * @apiParam {String} moduleId Module id
+   */
+  router.post(
+    {
+      path: '/api/ml/modules/setup/{moduleId}',
+      validate: {
+        params: schema.object({
+          moduleId: schema.string(),
+        }),
+        body: schema.object({
```

Comment:
What do you think about moving this to a schema file to be consistent with the other converted routes?
There are also enough schema files in the `new_platform` directory that I think it warrants creating a `schemas` dir in there. I'm happy to do that part but would be good to get these `body` schemas in a separate file.

Reply:
Moved to a dedicated file in https://github.com/elastic/kibana/pull/57206/commits/63ea75b825f7659b09b62f62ab420f33651637ed